x-pack/plugins/enterprise_search/public/applications/app_search/components/search/search_logic.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+
+import { Result } from '../result/types';
+
+interface SearchValues {
+  dataLoading: boolean;
+  searchQuery: string;
+  searchResults: Result[];
+}
+
+interface SearchActions {
+  search(query: string): { query: string };
+  onSearch({ results }: { results: Result[] }): { results: Result[] };
+}
+
+export const SearchLogic = kea<MakeLogicType<SearchValues, SearchActions>>({
```

Comment:
`SearchLogic` is a new logic file which can be reused across multiple components.

Currently, it is only the query tester and the add curations.

It maintains a "searchQuery" string in state, which can be updated using the "search" action, which then stores the "searchResults".

This can be consumed in a component as-is directly, or could even be extended using a Kea extend.